tools/add_triage_signature.py

```diff
@@ -1120,6 +1120,96 @@ def _update_ticket(self, url, issue_key, should_update=False):
+    writing_image_end_event_regex = re.compile(r'.*reached installation stage Writing image to disk: 100%$')
+    slow_threshold_seconds = 300
+
+    class NoEventFound(Exception):
+        pass
+
+    def __init__(self, jira_client):
+        super().__init__(jira_client, comment_identifying_string="h1. OS Installation Time:")
+
+    @classmethod
+    def _get_last_event(cls, all_events, regex):
+        """
+        We want to get the last event because sometimes the list of all events
+        for a triage ticket contains events from previous installations. The
+        last ones however, are always relevant to the ticket in question.
+        """
+        try:
+            *_, last = (
+                event
+                for event in all_events
+                if regex.match(event["message"]) is not None
+            )
+        except ValueError:
+            raise cls.NoEventFound
+
+        return last
+
+    @classmethod
+    def _get_last_event_timestamp(cls, all_events, regex):
+        return dateutil.parser.isoparse(cls._get_last_event(all_events, regex)["event_time"])
+
+    @classmethod
+    def _get_start_event_timestamp(cls, all_events):
+        return cls._get_last_event_timestamp(all_events, cls.writing_image_start_event_regex)
+
+    @classmethod
+    def _get_end_event_timestamp(cls, all_events):
+        return cls._get_last_event_timestamp(all_events, cls.writing_image_end_event_regex)
+
+    def _update_ticket(self, url, issue_key, should_update=False):
+        url = self._logs_url_to_api(url)
+        md = get_metadata_json(url)
+        cluster = md['cluster']
+        cluster_id = cluster['id']
+
+        events_by_host = defaultdict(list)
+        for event in get_events_json(cluster_url=url, cluster_id=cluster_id):
+            if "host_id" not in event:
+                # Cluster-level event, no host_id associated with it
+                continue
+            events_by_host[event["host_id"]].append(event)
+
+        hosts = []
+        for host in cluster['hosts']:
+            host_events = events_by_host[host["id"]]
+
+            try:
+                total_duration_seconds = (
+                    self._get_end_event_timestamp(host_events)
+                    -
+                    self._get_start_event_timestamp(host_events)
+                ).total_seconds()
+            except self.NoEventFound:
+                return
+
+            if total_duration_seconds <= 0:
+                # Negative writing to disk duration. This could happen if writing
+                # to disk never completed in this ticket but it did complete in
+                # previous runs.
+                return
+
+            if total_duration_seconds > self.slow_threshold_seconds:
+                host_disk_duration_seconds_message = (
+                    f"{{color:red}}OS installation to disk was rather slow, it took {total_duration_seconds} seconds{{color}}"
```

Comment:
I wonder if we could have the following logic (but treat it as a suggestion, I'm perfectly fine if you don't follow with it)

* if every host in the cluster installed faster than `slow_threshold_seconds`, don't print anything
* if at least one host in the cluster installed slower than `slow_threshold_seconds`, print `total_duration_seconds`

The latter is because if that happened that e.g. 1 out of 3 hosts exceeded a threshold, I would be interested to see what were values for the other hosts. For various reasons the following 2 scenarios are different

* 1 host installed in 15 minutes; 2 hosts installed in 1 minute
* 1 host installed in 5.5 minute; 2 hosts installed in 4.5 minute